internal/exec/scalar.go

```diff
@@ -36,15 +36,24 @@ var intScalar = &scalar{
 			}
 			return coerced, nil
 		default:
-			return nil, fmt.Errorf("wrong type")
+			return nil, fmt.Errorf("wrong type: %T", input)
```

Comment:
Why do you add the type here? It is already included in the error message by the code that calls this. Have you ever tested it?